azure-pipelines.yml

```diff
@@ -64,14 +64,14 @@ stages:
         Linux_Gcc_Release:
           image: ${{ variables.linux }}
           configuration: Release
-          CC: gcc-7
-          CXX: g++-7
+          CC: gcc-9
```

Comment:
a reason not to use gcc-12? (latest seems to be jammy, which seems to have gcc-12)

Reply:
gcc 12 is not natively available on this image. GCC 10 is and I could easily be convinced to switch to that. 

In both compiler and OS version, I'm trying to balance the eagerness to benefit from the advantages of the latest versions with the interest of supporting people who might not have the latest versions.

The highest available Ubuntu right now is this: https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md which has GCC 11.  However, to get it, we'd have to specify it by number and lose the auto updating advantage of just specifying -latest.